posthog/cdp/templates/reddit/template_reddit_pixel.py

```diff
@@ -0,0 +1,236 @@
+        "description": "Map of Reddit event attributes and their values. Check out these pages for more details: https://business.reddithelp.com/s/article/manual-conversion-events-with-the-reddit-pixel and https://business.reddithelp.com/s/article/about-event-metadata",
+        "label": "Event parameters",
+        "default": {
+            "conversion_id": "{event.uuid}",
+            "products": "{event.properties.products ? arrayMap(product -> ({'id': product.product_id, 'category': product.category, 'name': product.name}), event.properties.products) : event.properties.product_id ? [{'id': event.properties.product_id, 'category': event.properties.category, 'name': event.properties.name}] : undefined}",
+            "value": "{toFloat(event.properties.value ?? event.properties.revenue ?? event.properties.price)}",
+            "currency": "{event.properties.currency}",
+        },
+        "secret": False,
+        "required": False,
+    },
+]
+
+template_reddit_pixel: HogFunctionTemplate = HogFunctionTemplate(
+    free=True,
+    status="alpha",
+    type="site_destination",
+    id="template-reddit-pixel",
+    name="Reddit Pixel",
+    description="Track how many Reddit users interact with your website.",
+    icon_url="/static/services/reddit.png",
+    category=["Advertisement"],
+    hog="""
+// Adds window.rdt and lazily loads the Reddit Pixel script
+function initSnippet() {
+    !function(w,d){if(!w.rdt){var p=w.rdt=function(){p.sendEvent?p.sendEvent.apply(p,arguments):p.callQueue.push(arguments)};p.callQueue=[];var t=d.createElement("script");t.src="https://www.redditstatic.com/ads/pixel.js",t.async=!0;var s=d.getElementsByTagName("script")[0];s.parentNode.insertBefore(t,s)}}(window,document);
+}
+
+export function onLoad({ inputs }) {
+    initSnippet();
+    let userProperties = {};
+    for (const [key, value] of Object.entries(inputs.userProperties)) {
+        if (value) {
+            userProperties[key] = value;
+        }
+    };
+    if (posthog.config.debug) {
+        console.log('[PostHog.js] rdt init', inputs.pixelId, userProperties);
+    }
+    rdt('init', inputs.pixelId, userProperties);
+}
+export function onEvent({ inputs, posthog }) {
+    const builtInEvents = [
+        'PageVisit',
+        'Search',
+        'AddToCart',
+        'AddToWishlist',
+        'Purchase',
+        'ViewContent',
+        'Lead',
+        'SignUp',
+    ];
+    let eventProperties = {};
+    for (const [key, value] of Object.entries(inputs.eventProperties)) {
+        if (value) {
+            eventProperties[key] = value;
+        }
+    };
+    let eventName;
+    if (builtInEvents.includes(inputs.eventType)) {
+        eventName = inputs.eventType;
+    } else {
+        eventName = 'Custom';
+        eventProperties.customEventName = inputs.eventType;
+    }
+    if (posthog.config.debug) {
+        console.log('[PostHog.js] rdt track', eventName, eventProperties);
+    }
+    rdt('track', eventName, eventProperties);
```

Comment:
We can remove the builtInEvents logic here, and use the eventType input from the mappings instead. For the custom event, we could have a mapping with a filter for `not in (builtInEvents list)`.

Reply:
`builtInEvents` are the events built into the `rdt` function, rather than the posthog destination, which is why I kept them separate.

E.g. right now we have $pageview -> PageVisit as a mapping in the template. PageVisit is supported by rdt, so we don't do the custom event branch

Hypothetically, the user could change the mapping to $pageview -> PageVisitFoo. Tt's not supported to call rdt with that value as the event name, so with the code as it is, we do the custom event logic, and don't make an invalid call to rdt

This means that from the user's POV, sending custom events is as easy as adding a new mapping from oldEventName -> newEventName and they don't need to worry about making invalid calls this way.

I can change the code to document this more clearly

Reply:
Done and added a test